static/app/components/searchSyntax/parser.tsx

```diff
@@ -666,6 +672,17 @@ export class TokenConverter {
     return null;
   };
 
+  /**
+   * Checks a filter against some non-grammar validation rules
+   */
+  checkFilterWarning = <T extends FilterType>(key: FilterMap[T]['key']) => {
+    if (key.type !== Token.KEY_SIMPLE) {
+      return null;
+    }
+    const keyName = getKeyName(key as TokenResult<Token.KEY_SIMPLE>);
```

Comment:
Doe the guard not correctly discriminate the key?

Reply:
ah yea, seems like the type is a little bit too complicated for TS to understand